ipc/src/server.rs

```diff
@@ -160,7 +160,7 @@ pub struct Server {
 impl Server {
 	/// Closes the server (waits for finish)
 	pub fn close(mut self) {
-		self.stop.take().unwrap().complete(());
+		self.stop.take().map(|stop| stop.send(()).unwrap_or_else(|_| { /* might be closed already */ } ));
```

Comment:
Why not `let _ = self.stop.take().map(|stop| stop.send(()));` or `let _ = self.stop.take().and_then(|stop| stop.send(()).ok());`?